src/xcode/ENA/ENA/Source/Services/PPAccessControl/Model/SRSError.swift

```diff
@@ -20,35 +32,70 @@ enum SRSError: Error, Equatable {
 	case srsSUBServerError
 	case srsSUB400
 	case srsSUB403
-	
-    var description: String {
-        switch self {
-        case .ppacError(let ppacError):
-            return "ppacError: \(ppacError.description)"
-        case .otpError(let otpError):
-            return "otpError: \(otpError.description)"
-        case .srsOTPClientError:
-            return "SRS_OTP_CLIENT_ERROR"
-        case .srsOTPNetworkError:
-            return "SRS_OTP_NO_NETWORK"
-        case .srsOTPServerError:
-            return "SRS_OTP_SERVER_ERROR"
-        case .srsOTP400:
-            return "SRS_OTP_400"
-        case .srsOTP401:
-            return "SRS_OTP_401"
-        case .srsOTP403:
-            return "SRS_OTP_403"
-        case .srsSUBClientError:
-            return "SRS_SUB_CLIENT_ERROR"
-        case .srsSUBNoNetwork:
-            return "SRS_SUB_NO_NETWORK"
-        case .srsSUBServerError:
-            return "SRS_SUB_SERVER_ERROR"
-        case .srsSUB400:
-            return "SRS_SUB_400"
-        case .srsSUB403:
-            return "SRS_SUB_403"
-        }
-    }
+}
+
+extension SRSError: ErrorCodeProviding {
+	var description: String {
+		switch self {
+		case .ppacError(let ppacError):
+			return "ppacError: \(ppacError.description)"
+		case .otpError(let otpError):
+			return "otpError: \(otpError.description)"
+		case .srsOTPClientError:
+			return "SRS_OTP_CLIENT_ERROR"
+		case .srsOTPNetworkError:
+			return "SRS_OTP_NO_NETWORK"
+		case .srsOTPServerError:
+			return "SRS_OTP_SERVER_ERROR"
+		case .srsOTP400:
+			return "SRS_OTP_400"
+		case .srsOTP401:
+			return "SRS_OTP_401"
+		case .srsOTP403:
+			return "SRS_OTP_403"
+		case .srsSUBClientError:
+			return "SRS_SUB_CLIENT_ERROR"
+		case .srsSUBNoNetwork:
+			return "SRS_SUB_NO_NETWORK"
+		case .srsSUBServerError:
+			return "SRS_SUB_SERVER_ERROR"
+		case .srsSUB400:
+			return "SRS_SUB_400"
+		case .srsSUB403:
+			return "SRS_SUB_403"
+		}
+	}
+}
+
+extension SRSError: SRSErrorAlertProviding {
+	var srsErrorAlert: SRSErrorAlert? {
+		switch self {
+		case .ppacError(let ppacError):
+			return ppacError.srsErrorAlert
+		case .otpError(let otpError):
+			return otpError.srsErrorAlert
+		case .srsOTPClientError:
+			return .callHotline
+		case .srsOTPNetworkError:
+			return .noNetwork
+		case .srsOTPServerError:
+			return .tryAgainLater
+		case .srsOTP400:
```

Comment:
better to use comma with multiple values as the return value is same?

Reply:
good